models/evc.py

```diff
@@ -1410,6 +1410,45 @@ def handle_link_down(self):
 
         return success
 
+    @staticmethod
+    def get_interface_from_switch(uni: UNI, switches: dict) -> Interface:
+        """Get interface from switch by uni"""
+        switch = switches[uni.interface.switch.dpid]
+        interface = switch.interfaces[uni.interface.port_number]
+        return interface
+
+    def handle_topology_update(self, switches: dict):
+        """Handle changes in the topology"""
+        interface_a = self.get_interface_from_switch(self.uni_a, switches)
+        interface_z = self.get_interface_from_switch(self.uni_z, switches)
+        active = self.is_uni_interface_active(interface_a, interface_z)
+
+        if self.is_active() != active:
+            if active:
+                self.activate()
+            else:
+                self.deactivate()
+            self.sync()
```

Comment:
We've got cases where `self.sync()` is having a `O(n)` time complexity in the worst case. This is not an issue with this PR, but it's something that in the future it should be improved, if you could map a `priority_low` just so we could use a bulk write that'd be great, in this iteration, I'll recommend that you simulate at least 300 EVCs having their UNI going down and check out the APM, so see if currently it's reasonable, it probably will be, since we won't have > 10**4 typically

Reply:
[Issue](https://github.com/kytos-ng/mef_eline/issues/356) created